### PR TITLE
Tweak settings dialog

### PR DESCRIPTION
**Pull request by RussKie**
Second round of tweaks to the Settings dialogs. The first is #5058.

Most notable changes proposed in this pull request:
- rename 'Git' to 'Git Paths' tab
- scaling fixes to 'Colors' and 'Git Paths' tabs
- deprecate and remove application icon configurations
- move 'Branch Order' setting to 'Appearance' settings page
- remove `ToolbarSettingsPage`
 
Screenshots before and after (if PR changes UI):
old:
image

new:
image

What did I do to test the code and ensure quality:
- manual tests at different scaling

Has been tested on (remove any that don't apply):
- Windows 10

### CODE REVIEW

**Comment by RussKie** [comment]
> Why remove Lemmings.cs? Easter egg...

We have too much random code with random or unused and untested behaviours. I wanted to remove it for some time but never got around until now.
Lemmings were entangled with icon styles and colors, so they had to do.

> If removed, remove the referenced images too.

Will do, somehow I missed this step.

> I see no reason to have Colors and Fonts as submenues.

To me the current arrangement of Settings looks very random. By grouping them I'm trying to add some order. I somewhat expect there will be more appearance-related settings in coming releases.
Right now I don't have a fully-formed idea how to re-arrange 'Advanced' and 'Detailed' settings and where to put the dashboard-related configurations. This warrants a separate discussion altogether.

> As Git Path is a submenu to Git, it should be called just Git (not too long name though). Same for Git Config

Did you mean `Paths` and `Config`?
Right now it is 'Git > Git' which to me looks odd, hence I propose 'Git > Git paths'. Happy to rename to 'Git > Paths'. 
I'll fix the casing.

> Why Commit dialog in Advanced?

That's the existing functionality. Something to look at.

**Comment by RussKie** [comment]
There is an old discussion
https://github.com/gitextensions/gitextensions/issues/1707 that is quite
relevant.

**Comment by gerhardol** [comment]
> Right now I don't have a fully-formed idea how to re-arrange 'Advanced' and 'Detailed' settings and where to put the dashboard-related configurations. This warrants a separate discussion altogether.

Agree

>> Why Commit dialog in Advanced?
> That's the existing functionality. Something to look at.

In 2.51
image

**Comment by RussKie** [comment]
/facepalm
Thank you, Gerhard, I totally missed it. The commit dialog shouldn't have
moved, I must have botched a MC resolution.

**Comment by RussKie** [comment]
I've renamed nodes under "Git" as follows:
image

. . .

The last commit addresses a behaviour I observed while testing things, to me it looks like a scaling bug in WinForms - `NumericUpDown` control's margins seems to get scaled beyond expectations when Windows is set to scale.
For example, if design is performed at 96dpi, NUD's margins are default (3), however at 200% scaling the render looks like this:
image
image

If I re-layout everything at 196dpi, and margins are persisted as scaled, everything looks as expected.

To combat this, I've created `DpiUtil.ScaleDefaultMargins` (happy to change the name, if desired) that resets supplied NUD's margins and scales them as necessary. 

Then everything looks as expected:
image
image

File: GitExtUtils/GitUI/DpiUtil.cs [hunk 1]
```diff
@@ -51,6 +51,16 @@ static DpiUtil()
         /// </summary>
         public static bool IsNonStandard => DpiX != 96 || DpiY != 96;
 
+        /// <summary>
+        /// BUG: looks like a bug in WinForms (at least .NET 4.6) - NumericUpDown's margin gets scaled beyond expectations
+        /// For example, if the design was done at 100% and Windows is set to scale to 200%, NUD's margins gets set at 96...
+        /// </summary>
+        /// <param name="control">The control that needs to have margins reset to default values and scaled appropriately.</param>
+        public static void ScaleDefaultMargins([NotNull] NumericUpDown control)
```
Comment (drewnoakes) [inline GitExtUtils/GitUI/DpiUtil.cs]: Can this be added to the control walking done in `AdjustForDpiScaling`?

**Comment by drewnoakes** [comment]
If the application icons are to be removed, then #3889 becomes obsolete.

**Comment by RussKie** [comment]
I did just that while on a plane. Found that there are few more controls,
like TextBox and Label need the same margin reset. However there are
usecases where automatic reset is harmful - e.g. Color settings tab starts
looking very squashed....

I will separate this commit (a rework actually) and push it separately.

**Comment by RussKie** [comment]
Thanks, I'll update the commit message.

**Comment by RussKie** [comment]
I removed margin adjustment commit, it will come separately

**Comment by RussKie** [comment]
Sorry, Drew, you have a MC now :)

**Pull request merged**